examples/platforms/efr32/radio.c

```diff
@@ -0,0 +1,864 @@
+                 error = kThreadError_NoAddress);
+
+    removeFromSrcMatchShortIndirect(entry);
+
+exit:
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+ThreadError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const uint8_t *aExtAddress)
+{
+    ThreadError error = kThreadError_None;
+    int8_t entry = -1;
+    (void)aInstance;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    entry = findSrcMatchExtEntry(aExtAddress);
+    otLogDebgPlat(sInstance, "Clear ExtAddr entry: %d", entry);
+
+    VerifyOrExit(entry >= 0 && entry < RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM,
+                 error = kThreadError_NoAddress);
+
+    removeFromSrcMatchExtIndirect(entry);
+
+exit:
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+void otPlatRadioClearSrcMatchShortEntries(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    otLogDebgPlat(sInstance, "Clear ShortAddr entries", NULL);
+
+    memset(srcMatchShortEntry, 0, sizeof(srcMatchShortEntry));
+}
+
+void otPlatRadioClearSrcMatchExtEntries(otInstance *aInstance)
+{
+    (void)aInstance;
+
+    otLogDebgPlat(sInstance, "Clear ExtAddr entries", NULL);
+
+    memset(srcMatchExtEntry, 0, sizeof(srcMatchExtEntry));
+}
+
+void RAILCb_IEEE802154_DataRequestCommand(RAIL_IEEE802154_Address_t *aAddress)
+{
+    if (sIsSrcMatchEnabled)
+    {
+        if (aAddress->length == RAIL_IEEE802154_LongAddress)
+        {
+            if (findSrcMatchExtEntry(aAddress->longAddress) >= 0)
+            {
+                RAIL_IEEE802154_SetFramePending();
+            }
+        }
+        else
+        {
+            if (findSrcMatchShortEntry(aAddress->shortAddress) >= 0)
+            {
+                RAIL_IEEE802154_SetFramePending();
+            }
+        }
+    }
+    else
+    {
+        (void)aAddress;
+        RAIL_IEEE802154_SetFramePending();
+    }
+}
+
+ThreadError otPlatRadioEnergyScan(otInstance *aInstance, uint8_t aScanChannel, uint16_t aScanDuration)
+{
+    (void)aInstance;
+    (void)aScanChannel;
+    (void)aScanDuration;
+    return kThreadError_NotImplemented;
+}
+
+void RAILCb_TxRadioStatus(uint8_t aStatus)
+{
+    switch (aStatus)
+    {
+    case RAIL_TX_CONFIG_CHANNEL_BUSY:
+        sTransmitError = kThreadError_ChannelAccessFailure;
+        sTransmitBusy = false;
+        break;
+
+    case RAIL_TX_CONFIG_TX_ABORTED:
+        sTransmitError = kThreadError_Abort;
+        sTransmitBusy = false;
+        break;
+    }
+}
+
+void RAILCb_TxPacketSent(RAIL_TxPacketInfo_t *aTxPacketInfo)
+{
+    (void)aTxPacketInfo;
+    sTransmitError = kThreadError_None;
+    sTransmitBusy = false;
+}
+
+void RAILCb_RxPacketReceived(void *aRxPacketHandle)
+{
+    RAIL_RxPacketInfo_t *rxPacketInfo;
+    uint8_t length;
+
+    rxPacketInfo = (RAIL_RxPacketInfo_t *)aRxPacketHandle;
+
+    if (rxPacketInfo == NULL ||
+        rxPacketInfo->appendedInfo.crcStatus == false ||
+        rxPacketInfo->appendedInfo.frameCodingStatus == false)
+    {
+        return;
```

Comment:
As per the [style guide](https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md), functions should only have one return statement.  Could you use the `VerifyOrExit` macro here instead?